app/src/main/kotlin/com/github/opensmsforwarder/data/mapper/Mapper.kt

```diff
@@ -44,4 +45,21 @@ class Mapper @Inject constructor() {
             recipientId = ruleEntity.recipientId,
             textRule = ruleEntity.rule
         )
+
+    fun toForwardingHistoryEntity(
+        recipient: Recipient,
+        time: Long,
+        message: String,
+        isForwardingSuccessful: Boolean,
+    ): ForwardingHistoryEntity =
+        ForwardingHistoryEntity(
+            date = time,
+            title = recipient.title,
+            forwardingType = recipient.forwardingType,
+            recipientPhone = recipient.recipientPhone,
+            senderEmail = recipient.senderEmail,
+            recipientEmail = recipient.recipientEmail,
+            message = message,
+            isForwardingSuccessful = isForwardingSuccessful
```

Comment:
My thoughts were to connect 2 tables RecupientEntity and ForwardingHistory Entity via relation (foreign key and primary key). It will give us:
- cascade deleting when clear history of forwardedhistory if we remove the recipient
- be more flexible constructing SQL queries
- no need for isForwardingSuccessful field to be duplicated in 2 tables

Reply:
Drop isForwardingSuccessful from RecipientEntity